### PR TITLE
refactor g2p jobs

### PR DESCRIPTION
**Pull request by JackTemaki**

### CODE REVIEW

**Review by michelwi — APPROVED** [review]

File: g2p/convert.py [hunk 1]
```diff
@@ -36,18 +43,28 @@ def run(self):
...
+
+    def __init__(self, iv_bliss_lexicon, g2p_output_lexicon, merge=True):
+        """
+        :param Path iv_bliss_lexicon: bliss lexicon as reference for the phoneme inventory
+        :param Path g2p_output_lexicon: from ApplyG2PModelJob.out_g2p_lexicon
+        :param bool merge: merge the g2p lexicon into the iv_bliss_lexicon instead of
+            only taking the phoneme inventory
+        """
+        self.iv_bliss_lexicon = iv_bliss_lexicon
+        self.g2p_output_lexicon = g2p_output_lexicon
```
Comment (curufinwe) [inline g2p/convert.py:58]: better: g2p_lexicon

File: g2p/train.py [hunk 2]
```diff
@@ -21,6 +27,19 @@ def __init__(
...
         g2p_python=None,
     ):
+        """
+
+        :param Path g2p_lexicon: g2p_lexicon for training, use BlissLexiconToG2PLexiconJob to generate a g2p_lexicon
+            from a bliss lexicon
+        :param int num_ramp_ups: number of global ramp-ups (n-gram-iness)
+        :param int min_iter: minimum iterations per ramp-up
+        :param int max_iter: maximum iteration sper ramp-up
+        :param str devel: passed as -d argument
```
Comment (curufinwe) [inline g2p/train.py:37]: percent of train lexicon held out as validation set

File: g2p/train.py [hunk 3]
```diff
@@ -21,6 +27,19 @@ def __init__(
...
+        :param str size_constrains: passed as -s argument
```
Comment (curufinwe) [inline g2p/train.py:38]: multigrams must have l1 ... l2 left-symbols and r1 ... r2 right-symbols

**Review by curufinwe — CHANGES_REQUESTED** [review]

File: g2p/apply.py [hunk 4]
```diff
@@ -27,15 +40,15 @@ def __init__(
...
 
-        self.g2p_lexicon = self.output_path("g2p.lexicon")
+        self.out_g2p_lexicon = self.output_path("g2p.lexicon")
 
     def tasks(self):
         yield Task("run", mini_task=True)
 
     def run(self):
-        with open(self.g2p_lexicon.get_path(), "wt") as out:
+        with open(self.out_g2p_lexicon.get_path(), "wt") as out:
```
Comment (christophmluscher) [inline g2p/apply.py:51]: wrap with `uopen`?

File: g2p/convert.py [hunk 5]
```diff
@@ -24,7 +31,7 @@ def run(self):
         with util.uopen(self.bliss_lexicon, "rt") as f:
             tree = ET.parse(f)
 
-        with open(self.g2p_lexicon.get_path(), "wt") as out:
+        with open(self.out_g2p_lexicon.get_path(), "wt") as out:
```
Comment (christophmluscher) [inline g2p/convert.py:34]: wrap with `uopen`?

File: g2p/convert.py (hunk 1 above)
Comment (JackTemaki) [reply g2p/convert.py:58]: Is this in the exact same format as the input to the G2P training? If yes then this makes sense to change. I was not sure if they might look different.

Comment (curufinwe) [reply g2p/convert.py:57]: Yes, they are the same

**Review by curufinwe — APPROVED** [review]

**Pull request merged**